agent/engine/docker_task_engine.go

```diff
@@ -1133,15 +1133,21 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	}
 
 	if execcmd.IsExecEnabledContainer(container) {
+		var herr error
 		tId, err := task.GetID()
 		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+			herr = &apierrors.HostConfigError{Msg: err.Error()}
 		}
 		err = engine.execCmdMgr.InitializeContainer(tId, container, hostConfig)
 		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+			herr = &apierrors.HostConfigError{Msg: err.Error()}
+		}
+		if herr != nil {
+			seelog.Warnf("Exec Agent initialization failed : %v . Continuing to start container without enabling exec feature.", herr)
```

Comment:
Good point. Added a check to see if exec agent init failed in RestartExecAgent and StartExecAgent methods. I did not disable exec for that container because in delete task, we do some cleaning if the task is exec enabled.